src/mono/wasm/runtime/CMakeLists.txt

```diff
@@ -27,6 +27,7 @@ target_link_libraries(dotnet
     ${MONO_ARTIFACTS_DIR}/libmono-profiler-aot.a
     ${MONO_ARTIFACTS_DIR}/libmono-profiler-browser.a
     ${NATIVE_BIN_DIR}/libSystem.Native.a
+    ${NATIVE_BIN_DIR}/libSystem.Globalization.Native.a
```

Comment:
I must admit I don't understand big picture. Could you please explain the consequences of this for browser ?

Reply:
Instead of building the ICU shim as part of the Mono runtime build, it moves the build to libs.native part of the build. Then it links the resulting static library back.

The end result should be roughly identical to the previous status quo.

There are several reasons for this change:
- Reuse of the build system in libs.native instead of maintaining a second copy in Mono build
- iOS-like platforms need the library for both Mono and NativeAOT, so building it as part of Mono runtime is problematic; it can be shared by both runtimes this way
- Browser/WASI are just changed to follow the same pattern to make the build system less complicated

Reply:
thanks!

Reply:
/cc @steveisok LibraryBuilder targets should also be aware of this change (although I suspect it doesn't matter to the library builder where the static lib came from, as long as it's there)

Reply:
from what I can see the LibraryBuilder doesn't need updating since it just takes all *.a files